scanner/src/main/kotlin/scanners/fossid/FossIdScanResults.kt

```diff
@@ -124,10 +125,11 @@ internal fun mapSnippetFindings(
                     if (expression == null) {
                         issues += FossId.createAndLogIssue(
                             source = "FossId",
-                            message = "Failed to map license '$it' as an SPDX expression."
+                            message = "Failed to map license '$it' as a SPDX expression. Mapping it using licenseRef-.",
```

Comment:
commit: Mapping to a `LicenseRef-*` in my view does not make sense at all if it is not clear what the semantics of the ID actually is. This seems to be the case here, which is why I propose to not do this.

I propose the workflow to be as follows:
1. unmapped license strings are flagged as hint. Optionally by a policy rule as must fix, e.g. error.
2. In order to fix the mapping, one has to manually define a "license mapping" similar to the declared license mapping curation.

What do you think?

Reply:
> As FossID is a black box, we don't know if it is a declared license or a detected license.

Wasn't it so that as a user, you can add licenses in the FossID web UI, and instruct FossID to use licenses from there for identified snippets? If so, I think it would be rather "declared" licenses.

Reply:
Yes in FossID you can add licenses and edit them a "License View".
Then, when a file is pending identification (i.e. has several matching snippets) you can:
- use the license of a snippet (without modification)
- manually assign a license of the "License View" of FossID

Therefore, I believe this license editor is NOT for snippets nor to curate snippets.

Reply:
@fviernau An opinion on my comment ?

Reply:
> After discussion with @MarcelBochtler we also think you are right, that the license refs are not suitable here.

So, can we change the approach to output an issue saying that some (parts of the) license string could not be mapped, instead of creating `License-Ref-` IDs from strings with unknown semantics?

> However, dealing with an unmapped license with a declared license mapping is not correct.

Agree. However, I believe a mechanism which is quite a bit similar could make sense.


> Therefore, here are our suggestions:
> 
>     * Change the severity in case of unmappeable license to `HINT`, to remove the flooding of report with snippet related issues..
> 
>     * When there is a possibility to do snippet curation with ORT, bump up the severity to `WARNING`. Then the user will fix the issue with a (hopefully global) snippet curation.

I'm fine with these suggestions, as long as the `License-Refs` construction gets dropped. 
As future perspective, I believe it also makes sense to add an `unmapped_license` property to the snippet model.

Reply:
I'm ok if you don't do the unmapped license property at all in scope of your task. I just wanted to have some common vision for the future.

Reply:
Maybe setting the snippet license to `NO_ASSERTION` plus flagging an issue that the license could not be mapped, would be reasonable?

Reply:
This is the current logic:
See https://github.com/bosch-io/oss-review-toolkit/blob/131fd5759440054ea77164713598c16b7abfefac/scanner/src/main/kotlin/scanners/fossid/FossIdScanResults.kt#L123-L133
and
https://github.com/bosch-io/oss-review-toolkit/blob/131fd5759440054ea77164713598c16b7abfefac/scanner/src/test/kotlin/scanners/fossid/FossIdLicenseMappingTest.kt#L94-L101

Reply:
The diff LGTM now. The commit message needs to be aligned with it.

Reply:
I adapted a bit the commit message to mention the `NO_ASSERTION`.
Should anything more in it be changed ?